examples/cfd/stokes_mgn/pi_inference.py

```diff
@@ -0,0 +1,525 @@
+            + 1 * loss_v
+            + 1 * loss_p
+            + 10 * loss_u_in
+            + 10 * loss_v_in
+            + 10 * loss_u_noslip
+            + 10 * loss_v_noslip
+            + 1 * loss_mom_u
+            + 1 * loss_mom_v
+            + 10 * loss_cont
+        )
+        self.optimizer.zero_grad()
+        loss.backward()
+        self.optimizer.step()
+
+        return (
+            loss_u,
+            loss_v,
+            loss_p,
+            loss_u_in,
+            loss_v_in,
+            loss_u_noslip,
+            loss_v_noslip,
+            loss_mom_u,
+            loss_mom_v,
+            loss_cont,
+        )
+
+    def validation(self):
+        self.model.eval()
+        with torch.no_grad():
+            x_int, y_int = self.coords[:, 0:1], self.coords[:, 1:2]
+            results_int = self.graph.forward({"x": x_int, "y": y_int})
+            pred_u, pred_v, pred_p = (
+                results_int["u"],
+                results_int["v"],
+                results_int["p"],
+            )
+            pred_u = pred_u.detach().cpu().numpy()
+            pred_v = pred_v.detach().cpu().numpy()
+            pred_p = pred_p.detach().cpu().numpy()
+            error_u = np.linalg.norm(self.ref_u - pred_u) / np.linalg.norm(self.ref_u)
+            error_v = np.linalg.norm(self.ref_v - pred_v) / np.linalg.norm(self.ref_v)
+            error_p = np.linalg.norm(self.ref_p - pred_p) / np.linalg.norm(self.ref_p)
+            self.wb.log(
+                {
+                    "test_u_error (%)": error_u,
+                    "test_v_error (%)": error_v,
+                    "test_p_error (%)": error_p,
+                }
+            )
+            return error_u, error_v, error_p
+
+
+if __name__ == "__main__":
+    # CUDA support
+    if torch.cuda.is_available():
+        device = torch.device("cuda")
+    else:
+        device = torch.device("cpu")
+
+    # initialize loggers
+    initialize_wandb(
+        project="Modulus-Launch",
+        entity="Modulus",
+        name="Stokes-Physics-Informed-Inference",
+        group="Stokes-DDP-Group",
+        mode=C.wandb_mode,
+    )
+
+    logger = PythonLogger("main")  # General python logger
+    logger.file_logging()
+
+    # Get dataset
+    path = os.path.join(C.results_dir, C.graph_path)
+
+    (
+        ref_u,
+        ref_v,
+        ref_p,
+        gnn_u,
+        gnn_v,
+        gnn_p,
+        coords,
+        coords_inflow,
+        coords_outflow,
+        coords_wall,
+        coords_polygon,
+        nu,
+    ) = get_dataset(path)
+    coords_noslip = np.concatenate([coords_wall, coords_polygon], axis=0)
+
+    # Initialize model
+    pi_inferencer = PhysicsInformedInferencer(
+        wb,
+        device,
+        gnn_u,
+        gnn_v,
+        gnn_p,
+        coords,
+        coords_inflow,
+        coords_noslip,
+        nu,
+        ref_u,
+        ref_v,
+        ref_p,
+    )
+
+    logger.info("Physics-informed inference started...")
```

Comment:
Infernce? The code below calls `train`?

Reply:
Thanks, please refer my response here: https://github.com/NVIDIA/modulus/pull/275#discussion_r1433020288

I have added more context to hopefully make it clearer.